pennylane/ops/op_math/pow.py

```diff
@@ -236,6 +236,15 @@ def z(self):
     def ndim_params(self):
         return self.base.ndim_params
 
+    @property
+    def data(self):
+        """The trainable parameters"""
+        return self.base.data
```

Comment:
This seems like an extremely stupid question to ask, but is `z` _never_ supposed to be trainable?

Reply:
I was wondering something similar in the above thread, but I'm not gonna question it in this PR.

Reply:
This allowed the base data to also be updated when `Pow`'s data was updated. We can reconsider this decision once parameter-shift no longer uses `QuantumScript.set_parameters` and in-place manipulation of an array.